src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -293,17 +304,26 @@ public Version getCreationVersion() {
     }
 
     /**
-     * Return the {@link Version} of that created the oldest segment in the index.
-     *
-     * If the index was created before v1.6 and didn't go through upgrade API the creation verion is returned.
-     * Returns null if the index was created before 0.19.0.RC1.
+     * Return the {@link Version} on which this index has been upgraded. This
+     * information is typically useful for backward compatibility.
+     */
+    public Version upgradeVersion() {
+        return indexUpgradedVersion;
+    }
+
+    public Version getUpgradeVersion() {
+        return upgradeVersion();
+    }
+
+    /**
+     * Return the {@link org.apache.lucene.util.Version} of the oldest lucene segment in the index
      */
-    public Version minimumCompatibleVersion() {
-        return indexMinimumCompatibleVersion;
+    public org.apache.lucene.util.Version minimumLuceneSegmentVersion() {
```

Comment:
Do we really need 2 functions that return the same thing? Especially with changing this name anyways?